### PR TITLE
upgrade django-tastypie

### PR DESCRIPTION
**Pull request by nickpell**
upgrades django-tastypie to remove deprecation warnings

:tropical_fish:

### CODE REVIEW

File: corehq/apps/api/urls.py [hunk 1]
```diff
@@ -73,9 +73,9 @@
         v0_5.StockTransactionResource,
         InternalFixtureResource,
         FixtureResource,
-        ReportResource,
+        ReportResource,#
```
Comment (dimagimon) [inline corehq/apps/api/urls.py]: E261 at least two spaces before inline comment

File: corehq/apps/api/urls.py [hunk 2]
```diff
@@ -73,9 +73,9 @@
...
         v0_5.DeviceReportResource,
-        DomainMetadataResource,
+        DomainMetadataResource, #
```
Comment (dimagimon) [inline corehq/apps/api/urls.py]: E261 at least two spaces before inline comment

**Comment by esoergel** [comment]
:no_entry_sign: :warning:

File: corehq/apps/api/tests.py [hunk 3]
```diff
@@ -966,12 +988,22 @@ def __init__(self, objs):
     def obj_get_list(self):
         return self.objs
 
+    def detail_uri_kwargs(self, bundle_or_obj):
+        return {
+            'pk': get_obj(bundle_or_obj).other_model_id
```
Comment (esoergel) [inline corehq/apps/api/tests.py]: Looks like this is `other_model_id`, whereas in the other places it's `other_model_ids` - is that intentional?  What do those represent, exactly?

Comment (esoergel) [inline corehq/apps/api/tests.py]: I see - it's a field on this test model.  I thought it was some really awkwardly named couchdbkit api I hadn't seen before.

**Pull request merged**